src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -3802,6 +3818,64 @@ SubqueryPushdownMultiNodeTree(Query *queryTree)
 }
 
 
+/*
+ * FlattenJoinVars iterates over provided columnList to identify
+ * Var's that are referenced from join RTE, and reverts back to their
+ * original RTEs.
+ *
+ * This is required because Postgres allows columns to be referenced using
+ * a join alias. Therefore the same column from a table could be referenced
+ * twice using its absolute table name (t1.a), and without table name (a).
+ * This is a problem when one of them is inside the group by clause and the
+ * other is not. Postgres is smart about it to detect that both target columns
+ * resolve to the same thing, and allows a single group by clause to cover
+ * both target entries. We want to make sure we provide correct varno/varattno
+ * values to Postgres so that it could produce valid query.
+ *
+ * Only exception is that, if a join is given an alias name, we do not want to
+ * flatten those var's.
```

Comment:
Maybe very briefly mention that in case joinalias exists, the inner rtes cannot be referenced? (Is that the reason, right?)

Reply:
correct